src/Helpers/PageHelpers/PageHelper.cs

```diff
@@ -96,6 +106,28 @@ public class PageHelper : IPageHelper
             return formModel;
         }
 
+        private async Task AddDynamicOptions(IElement element, FormAnswers formAnswers)
+        {
+            var submitDetails = element.Properties.Lookup
+                .SingleOrDefault(x => x.EnvironmentName.Equals(_environment.EnvironmentName));
+
+            Answers query = formAnswers.AllAnswers
+                .SingleOrDefault(x => x.QuestionId.Equals(element.Properties.LookupQuestionIdQueryKey));
+
+            if (string.IsNullOrEmpty((string)query?.Response) || string.IsNullOrEmpty(submitDetails.Provider))
+                throw new Exception("Dynamic lookup: No Query Details Found.");
+
+            var lookupProvider = _lookupProviders.Get(submitDetails.Provider);
+            if(lookupProvider == null)
+                throw new Exception("Dynamic lookup: No Lookup Provider Found.");
+
+            var lookupOptions = await lookupProvider.GetAsync(submitDetails.URL += (string)query.Response, submitDetails.AuthToken);
```

Comment:
either pass the SubmitSlug object or use string interpolation on the first paramter, also assumes the url will end with the query string?, should it be able to support http://localhost/api/v1/Data/{blah}/account